### PR TITLE
Add dac

### PR DESCRIPTION
**Pull request by bentwire**
Stole the Dac code from the STM32G0xx HAL crate and modified it to work with the STM32F072.  Only tested with 072 so only the 072 feature has the driver right now.

### CODE REVIEW

**Comment by bentwire** [comment]
Looks like I need to fix feature gating...

**Comment by bentwire** [comment]
Fixed!

File: src/dac.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+pub struct C2;
+
+pub trait DacOut<V> {
+    fn set_value(&mut self, val: V);
+    fn get_value(&mut self) -> V;
+}
+
+pub trait DacPin {
+    fn enable(&mut self);
+//    fn calibrate(&mut self, delay: &mut Delay);
```
Comment (therealprof) [inline src/dac.rs]:
```suggestion
```

Comment (bentwire) [reply src/dac.rs]: oops!  I will get that.

**Comment by bentwire** [comment]
Ok I will fix this ASAP, probably this weekend. Thanks!

**Comment by bentwire** [comment]
Er forgot to hit the comment button on the last comment...  So the feature gates and CHANGELOG entry are added.  Just need to add the example.

**Comment by bentwire** [comment]
Oh and docs.

**Comment by therealprof** [comment]
Oh, and I just noticed it doesn't compile, most likely because because you need to feature gate the example for stm32f072, you can check out `Cargo.toml` on how to do that.

**Comment by bentwire** [comment]
Hopefully this last push should fix those issues. :)  Just lost my job, have more time to work on this now!

**Pull request merged**